### PR TITLE
- FIX Uncaught Error: Call to a member function getStatusCode() on null

### PR DESCRIPTION
**Pull request by spidgorny**
- FIX Uncaught Error: Call to a member function getStatusCode() on null

### CODE REVIEW

File: lib/Github/Api/AbstractApi.php [hunk 1]
```diff
@@ -74,8 +75,11 @@ protected function get($path, array $parameters = array(), $requestHeaders = arr
             unset($parameters['ref']);
         }
         $response = $this->client->getHttpClient()->get($path, $parameters, $requestHeaders);
-
-        return ResponseMediator::getContent($response);
+        if ($response) {
+			return ResponseMediator::getContent($response);
+		} else {
+        	throw new \InvalidArgumentException(__METHOD__);
+		}
```
Comment (cursedcoder) [inline lib/Github/Api/AbstractApi.php]: style is broken here, see how to fix it https://styleci.io/analyses/qBwZZ7

File: lib/Github/Api/AbstractApi.php [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 namespace Github\Api;
 
+use Geocoder\Exception\InvalidArgument;
```
Comment (acrobat) [inline lib/Github/Api/AbstractApi.php]: I think you've imported an incorrect class `Geocoder\Exception\InvalidArgument`

File: lib/Github/Api/AbstractApi.php [hunk 3]
```diff
@@ -74,8 +75,11 @@ protected function get($path, array $parameters = array(), $requestHeaders = arr
             unset($parameters['ref']);
         }
         $response = $this->client->getHttpClient()->get($path, $parameters, $requestHeaders);
-
-        return ResponseMediator::getContent($response);
+        if ($response) {
```
Comment (Nyholm) [inline lib/Github/Api/AbstractApi.php]: In what scenario is there no response here?

Comment (Nyholm) [inline lib/Github/Api/AbstractApi.php]: I think we should analyze this in detail. I suspect that this PR does not solve the issue. 

Why does it only happen for GET requests? If this was **the** solution, shouldn't it be applied for all HTTP methods?

Are you using caching when you get this error?

File: lib/Github/HttpClient/CachedHttpClient.php [hunk 4]
```diff
@@ -115,7 +115,7 @@ protected function createRequest($httpMethod, $path, $body = null, array $header
     public function getLastResponse($force = false)
     {
         $lastResponse =  parent::getLastResponse();
-        if (304 != $lastResponse->getStatusCode()) {
+        if ($lastResponse && 304 != $lastResponse->getStatusCode()) {
```
Comment (Nyholm) [inline lib/Github/HttpClient/CachedHttpClient.php]: However, I agree with this change. It should be made in a separate PR since it is not related to the other changes.

Comment (Nyholm) [inline lib/Github/HttpClient/CachedHttpClient.php]: I saw that these changes are for the 1.7 branch. 
I believe that this line only will fix the issue addressed. 

@spidgorny Can you provide a stacktrace of the exception?

**Comment by Nyholm** [comment]
I will close this issue if there is no response before the end of April.

**Comment by Nyholm** [comment]
Thanks.
